memberships/tests.py

```diff
@@ -0,0 +1,63 @@
+from django.test import TransactionTestCase
+from django.urls import reverse
+from django.core.exceptions import ValidationError
+from django.contrib.auth.models import User
+
+from .forms import RegistrationForm
+from .models import Member
+
+
+class RegisterFormTestCase(TransactionTestCase):
+    def test_correct_fields_are_required(self):
+        response = self.client.post(reverse("register"))
+        self.assertFormError(response, "form", "full_name", "This field is required.")
+        self.assertFormError(response, "form", "email", "This field is required.")
+        self.assertFormError(response, "form", "password", "This field is required.")
+        self.assertFormError(response, "form", "birth_date", "This field is required.")
+        self.assertFormError(
+            response, "form", "constitution_agreed", "This field is required."
+        )
+
+    def test_preferred_name_defaults_to_full_name(self):
+        self.client.post(
+            reverse("register"),
+            {
+                "full_name": "test person",
+                "email": "test@person.com",
```

Comment:
Being really picky but can we test with example.com please? We know that will never resolve to anything other than the example site, unlike this particular domain.

Reply:
Sorted.